warning: declaration of ‘foobar’ shadows a member/a previous local

When compiling nan with -Wshadow I get warnings:

```
../node_modules/nan/nan.h: In constructor ‘NanAsyncWorker::NanAsyncWorker(NanCallback*)’:
../node_modules/nan/nan.h:1858:7: warning: declaration of ‘callback’ shadows a member of 'this' [-Wshadow]
       : callback(callback), errmsg_(NULL) {
       ^
In file included from ../src/pcsclite.h:4:0,
                 from ../src/pcsclite.cpp:1:
../node_modules/nan/nan.h: In constructor ‘NanAsyncWorker::NanAsyncWorker(NanCallback*)’:
../node_modules/nan/nan.h:1858:7: warning: declaration of ‘callback’ shadows a member of 'this' [-Wshadow]
       : callback(callback), errmsg_(NULL) {
       ^
In file included from ../src/cardreader.h:4:0,
                 from ../src/cardreader.cpp:1:
../node_modules/nan/nan.h: In constructor ‘NanAsyncWorker::NanAsyncWorker(NanCallback*)’:
../node_modules/nan/nan.h:1858:7: warning: declaration of ‘callback’ shadows a member of 'this' [-Wshadow]
       : callback(callback), errmsg_(NULL) {
       ^
../node_modules/nan/nan.h: In function ‘void* _NanRawString(v8::Handle<v8::Value>, Nan::Encoding, size_t*, void*, size_t, int)’:
../node_modules/nan/nan.h:2179:16: warning: declaration of ‘len’ shadows a previous local [-Wshadow]
         size_t len = toStr->Write(twobytebuf, 0,
                ^
../node_modules/nan/nan.h:2138:10: note: shadowed declaration is here
   size_t len;
          ^
../node_modules/nan/nan.h: In function ‘void* _NanRawString(v8::Handle<v8::Value>, Nan::Encoding, size_t*, void*, size_t, int)’:
../node_modules/nan/nan.h:2179:16: warning: declaration of ‘len’ shadows a previous local [-Wshadow]
         size_t len = toStr->Write(twobytebuf, 0,
                ^
../node_modules/nan/nan.h:2138:10: note: shadowed declaration is here
   size_t len;
          ^
../node_modules/nan/nan.h: In function ‘void* _NanRawString(v8::Handle<v8::Value>, Nan::Encoding, size_t*, void*, size_t, int)’:
../node_modules/nan/nan.h:2179:16: warning: declaration of ‘len’ shadows a previous local [-Wshadow]
         size_t len = toStr->Write(twobytebuf, 0,
                ^
../node_modules/nan/nan.h:2138:10: note: shadowed declaration is here
   size_t len;
          ^
```

--- a/nan.h
+++ b/nan.h
@@ -1865,8 +1865,8 @@ class NanCallback {
 
 /* abstract */ class NanAsyncWorker {
  public:
-  explicit NanAsyncWorker(NanCallback *callback)
-      : callback(callback), errmsg_(NULL) {
+  explicit NanAsyncWorker(NanCallback *callback_)
+      : callback(callback_), errmsg_(NULL) {
     request.data = this;
 
     NanScope();
@@ -2187,15 +2187,15 @@ namespace Nan {
       {
         uint16_t* twobytebuf = new uint16_t[sz_ + term_len];
 
-        size_t len = toStr->Write(twobytebuf, 0,
+        size_t somelen = toStr->Write(twobytebuf, 0,
           static_cast<int>(sz_ + term_len), flags);
 
-        for (size_t i = 0; i < sz_ + term_len && i < len + term_len; i++) {
+        for (size_t i = 0; i < sz_ + term_len && i < somelen + term_len; i++) {
           unsigned char *b = reinterpret_cast<unsigned char*>(&twobytebuf[i]);
           to[i] = *b;
         }
 
-        NanSetPointerSafe<size_t>(datalen, len);
+        NanSetPointerSafe<size_t>(datalen, somelen);
 
         delete[] twobytebuf;
         return to;

